src/platform/user/profile/vet360/components/AddressField/AddressEditModal.jsx

```diff
@@ -113,7 +113,7 @@ class AddressEditModal extends React.Component {
     const newAddressValue = { ...this.props.field.value, ...mailingAddress };
     if (useNewAddressForm) {
       this.props.onChangeFormDataAndSchemas(
-        newAddressValue,
+        this.transformInitialFormValues(newAddressValue),
```

Comment:
this cleans up the mailing address (removes null values) before using it as the residential address.